src/main/scala/where/Predicate.scala

```diff
@@ -1,10 +1,32 @@
 package com.originate.scalypher.where
 
 import com.originate.scalypher.ConstantString
+import com.originate.scalypher.path.Path
 import com.originate.scalypher.ToQuery
 
-sealed trait Predicate extends ToQuery
+case class PredicateWithProjection(
+  predicate: Predicate,
+  projection: PathProjection
+) {
+  def where(f: ObjectReference => Where): PredicateCondition =
```

Comment:
I think you can cover this nicely by adding a type parameter on `PredicateWithProjection` to propagate what you're working with exactly, then using that here:
```
case class PredicateWithProjection[Ref]( ... ) {
  def where(f: Ref => Where): PredicateCondition[Ref] = ...

...
sealed trait Predicate extends ToQuery {
  def nodesIn(path: Path): PredicateProjection[Seq[Node]] = ...
  def nodeIn(path: Path): PredicateProjection[Node] = ...
  // etc
}
```